File: packages/flic/ios/Plugin/Classes/Events/ScanStatusChangedEvent.swift
import Foundation
import Capacitor

@objc public class ScanStatusChangedEvent: NSObject, Result {
    let status: ScanStatus

    init(status: ScanStatus) {
        self.status = status
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["status"] = status.rawValue
        return result as AnyObject
    }
}

File: packages/flic/android/src/main/java/io/capawesome/capacitorjs/plugins/flic/classes/events/ScanStatusChangedEvent.java
package io.capawesome.capacitorjs.plugins.flic.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.flic.enums.ScanStatus;
import io.capawesome.capacitorjs.plugins.flic.interfaces.Result;

public class ScanStatusChangedEvent implements Result {

    @NonNull
    private final ScanStatus status;

    public ScanStatusChangedEvent(@NonNull ScanStatus status) {
        this.status = status;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("status", status.getValue());
        return result;
    }
}
